### PR TITLE
refactor: improve previews abstraction

### PR DESCRIPTION
**Pull request by wmalik**
## What this PR does / why we need it:

This PR addresses the concerns raised in the following PR:

* https://github.com/terramate-io/terramate/pull/1436

## Does this PR introduce a user-facing change?
```
No.

```

### CODE REVIEW

**Review by i4ki — APPROVED** [review]
LGTM

**Review by i4ki — APPROVED** [review]
LGTM

File: cmd/terramate/cli/run.go [hunk 1]
```diff
@@ -454,14 +453,16 @@ func (c *cli) createCloudPreview(runs []runContext) map[string]string {
...
-		prUpdatedAt = eventPRUpdatedAt.Unix()
+	githubEventPath, ok := os.LookupEnv("GITHUB_EVENT_PATH")
+	if ok {
+		eventPRUpdatedAt := github.GetEventPRUpdatedAt(githubEventPath)
+		if eventPRUpdatedAt != nil {
+			prUpdatedAt = eventPRUpdatedAt.Unix()
+		}
+	} else {
+		printer.Stderr.Warn(
+			sprintf("env var GITHUB_EVENT_PATH not found, using %d as updated_at", prUpdatedAt))
```
Comment (mariux) [inline cmd/terramate/cli/run.go:465]: We can not fall back here, we will need the event.

**Review by mariux — DISMISSED** [review]
we can not fall back when there is no event payload.

File: cloud/cloud_preview_test.go [hunk 2]
```diff
@@ -1,9 +1,13 @@
 // Copyright 2024 Terramate GmbH
 // SPDX-License-Identifier: MPL-2.0
 
-package previews_test
+package cloud_test
```
Comment (mariux) [inline cloud/cloud_preview_test.go:4]:
```suggestion
package cloud_preview_test
```

File: cmd/terramate/cli/run.go (hunk 1 above)
Comment (i4ki) [reply cmd/terramate/cli/run.go:465]: we discussed in a long call today about introducing a `TM_TEST_IGNORE_GITHUB_EVENT_FILE` (that's only available in a test build) for making it easy to locally test this and then fallback will happen only in this case.
We agree that a silent fallback here is not correct.

I think @wmalik will introduce this test env in the new PR.

**Pull request merged**